`golang` version `1.20` on `MacOS` ARM processors generates `status-go-library` with missing symbols

# Problem

`make test-contract` on MacOS fails with : 

```
dyld[30877]: missing symbol called
```

The problem initially showcased itself after the `golang` `1.20` upgrade was merged.
ref -> https://github.com/status-im/status-mobile/issues/19802

To fix this problem quickly I had modified the `status-go-library` derivation to build with `goPackage` `1.19` only for `Darwin` 
ref -> https://github.com/status-im/status-mobile/pull/19965

However a PR last week at `status-go` bumped `go-waku` version and along with it a bunch of libraries that are incompatible with `golang` 1.19, specifically `quic-go`
ref ->  https://github.com/status-im/status-go/pull/5150

Here are possible solutions to the current problem: 
- Ask Devs to stop developing on MacOS OR
- revert the status-go PR OR
- find a proper fix for the missing symbol issue.

note : further upgrading go to 1.21 or 1.22 may or may not fix this issue.

right away I see something suspicious : 

```
"Cannot find module 'bufferutil'
Require stack:
/Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/lib/buffer-util.js
/Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/lib/permessage-deflate.js
/Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/lib/websocket.js
/Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/index.js
/Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/@walletconnect/jsonrpc-ws-connection/dist/index.cjs.js
/Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/@walletconnect/core/dist/index.cjs.js
/Users/siddarthkumar/code/status-im/PR/status-mobile/target/contract_test/test.js"

```
and

```
"Cannot find module 'utf-8-validate'
Require stack:
- /Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/lib/validation.js
- /Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/lib/receiver.js
- /Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/lib/websocket.js
- /Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/ws/index.js
- /Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/@walletconnect/jsonrpc-ws-connection/dist/index.cjs.js
- /Users/siddarthkumar/code/status-im/PR/status-mobile/node_modules/@walletconnect/core/dist/index.cjs.js
- /Users/siddarthkumar/code/status-im/PR/status-mobile/target/contract_test/test.js"
```

--- a/nix/status-go/library/default.nix
+++ b/nix/status-go/library/default.nix
@@ -22,11 +22,14 @@ buildGoPackage {
   '';
 
   # Build the Go library
+  # ld flags and netgo tag are necessary for integration tests to work on MacOS
+  # https://github.com/status-im/status-mobile/issues/20135
   buildPhase = ''
     runHook preBuild
     go build \
       -buildmode='c-archive' \
-      -tags='gowaku_skip_migrations gowaku_no_rln' \
+      -ldflags '-w -s -extldflags "-lresolv"' \
+      -tags='gowaku_skip_migrations gowaku_no_rln netgo' \
       -o "$out/libstatus.a" \
       $NIX_BUILD_TOP/main.go
     runHook postBuild

--- a/nix/status-go/library/default.nix
+++ b/nix/status-go/library/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, meta, source, buildGoPackage }:
+{ lib, stdenv, meta, source, buildGoPackage }:
 
 buildGoPackage {
   pname = source.repo;
@@ -28,8 +28,8 @@ buildGoPackage {
     runHook preBuild
     go build \
       -buildmode='c-archive' \
-      -ldflags '-w -s -extldflags "-lresolv"' \
-      -tags='gowaku_skip_migrations gowaku_no_rln netgo' \
+      ${lib.optionalString stdenv.isDarwin "-ldflags=-extldflags=-lresolv"} \
+      -tags='gowaku_skip_migrations gowaku_no_rln ${lib.optionalString stdenv.isDarwin "netgo"}' \
       -o "$out/libstatus.a" \
       $NIX_BUILD_TOP/main.go
     runHook postBuild

--- a/nix/status-go/library/default.nix
+++ b/nix/status-go/library/default.nix
@@ -1,9 +1,6 @@
-{ stdenv, meta, source, buildGo119Package, buildGo120Package }:
-let
-  # https://github.com/status-im/status-mobile/issues/19802
-  # only for Darwin to fix Integration Tests failing with missing symbols on go 1.20
-  buildGoPackageIntegrationTest = if stdenv.isDarwin then buildGo119Package else buildGo120Package;
-in buildGoPackageIntegrationTest {
+{ stdenv, meta, source, buildGoPackage }:
+
+buildGoPackage {
   pname = source.repo;
   version = "${source.cleanVersion}-${source.shortRev}";
 

